modules/core/src/main/scala/org/tesselation/domain/node/NodeStorage.scala

```diff
@@ -0,0 +1,24 @@
+package org.tesselation.domain.node
+
+import cats.Applicative
+
+import org.tesselation.schema.node.NodeState
+
+trait NodeStorage[F[_]] {
+  def getNodeState: F[NodeState]
+
+  def setNodeState(nodeState: NodeState): F[Unit]
+
+  def tryModifyState[A](from: Set[NodeState], onStart: NodeState, onFinish: NodeState, fn: => F[A]): F[A]
```

Comment:
I am wondering if it would be useful or cleaner to curry 🤔 

`def tryModifyState[A](from, onStart, onFinish)(fn): F[A]`

Reply:
yep, great idea. let me try